packages/react-bootstrap-table2-filter/src/wrapper.js

```diff
@@ -40,33 +40,42 @@ export default (Base, {
       }
     }
 
-    onFilter(column, filterVal, filterType) {
-      const { store, columns } = this.props;
-      const currFilters = Object.assign({}, this.state.currFilters);
-      const { dataField, filter } = column;
+    /**
+     * filter the table like below:
+     * onFilter(column, filterType)(filterVal)
+     * @param {Object} column
+     * @param {String} filterType
+     * @param {String} filterVal - user input for filtering.
+     */
+    onFilter(column, filterType) {
+      return (filterVal) => {
```

Comment:
wrapper `onFilter` in `HOF` to allow user to set filter value programmatically.

Reply:
hey @Chun-MingChen after review, I think just one question here:

Why need to defined a HoF here?, I think even if we don't use HoF, the user still can call filter function programmatically, right?  you can check any filter component's `componentDidMount` function:

```js
if (getFilter) {
+      getFilter((filterVal) => {
+        this.setState(() => ({ isSelected: filterVal !== '' }));
+        this.selectInput.value = filterVal;

          // onFilter(column, filterVal, FILTER_TYPE.SELECT) can be work, right?
+        onFilter(column, FILTER_TYPE.SELECT)(filterVal);
+      });
     }
```

Any thought can shared with me?

Reply:
Hi @AllenFang,

Sorry for replying so late. The following is the first edition about how implemented programmatically filter. Actually I planned to return `onFilter` only. In view of UI state, I exported the anonymous function including `setState` and `onFilter` finally.

```js
// First edition, export onFilter function only to allow users to access
if (getFilter) {
  getFilter(
    onFilter(column, FILTER_TYPE.TEXT)
  );
}
```

I thought it became more concise and it might have no performance impact so I kept in the same way. Any concern please let me know and let's get further discussion. :)